packages/@react-aria/overlays/src/calculatePosition.ts

```diff
@@ -425,7 +477,14 @@ export function calculatePosition(opts: PositionOpts): PositionResult {
   let scrollSize = getScroll(scrollNode);
   let boundaryDimensions = getContainerDimensions(boundaryElement);
   let containerDimensions = getContainerDimensions(container);
+  // If the container is the HTML element wrapping the body element, the retrieved scrollTop/scrollLeft will be equal to the
+  // body element's scroll. Set the container's scroll values to 0 since the overlay's edge position value in getDelta don't then need to be further offset
+  // by the container scroll since they are essentially the same containing element and thus in the same coordinate system
   let containerOffsetWithBoundary: Offset = boundaryElement.tagName === 'BODY' ? getOffset(container) : getPosition(container, boundaryElement);
+  if (container.tagName === 'HTML' && boundaryElement.tagName === 'BODY') {
+    containerDimensions.scroll.top = 0;
+    containerDimensions.scroll.left = 0;
+  }
```

Comment:
This feels pretty gross, I was operating under the assumption that the boundary element would contain the container with all the changes in this code to calculate overlay/boundary edges but the above makes it a bit iffy. I'm not 100% sure this covers all the possible cases where perhaps the body is offset from the HTML element but the cases I tested with absolute positioning/applying a margin on the body seem to work alright. If anything it really feels like we should move over to floating-ui haha

Reply:
Is this related the the webkit issue you mentioned above? I didn't quite follow the description...

Reply:
Not quite, the webkit one is the following in getContainerDimensions:
```
  if (isWebKit() && (containerNode.tagName === 'BODY' || containerNode.tagName === 'HTML') && isPinchZoomedIn) {
    // Safari will report a non-zero scrollTop/Left for the non-scrolling body/HTML element when pinch zoomed in unlike other browsers.
    // Set to zero for parity calculations so we get consistent positioning of overlays across all browsers.
    // Also switch to visualViewport.pageTop/pageLeft so that we still accomodate for scroll positioning for body/HTML elements that are actually scrollable
    // before pinch zoom happens
    scroll.top = 0;
    scroll.left = 0;
    top = visualViewport.pageTop;
    left = visualViewport.pageLeft;
  }
```

this one is mainly highlighting that the base case where boundary = 'body' and container = 'html' is different from the envisioned typical setup of an arbitrary boundary element that contains an arbitrary container element which has the trigger element since the container in the former case is actually around the boundary element instead AND typically is of the same dimensions. Since those two are typically the same "box" their coordinate systems should be the same (aka 0,0 refers to the same point from the view point of the boundary and the container) and thus the overlay's top/left values is already in the same coordinate system as the boundary and thus shouldn't need to be offset by the container's scroll. This means we need `containerDimensions` scroll value to be 0 in that case so that the following code in `getDelta` doesn't calculate the wrong `startEdgeOffset`/`endEdgeOffset` values
```
  let containerScroll = containerDimensions.scroll[axis];
  // The height/width of the boundary. Matches the axis along which we are adjusting the overlay position
  let boundarySize = boundaryDimensions[AXIS_SIZE[axis]];
  // Calculate the edges of the boundary (accomodating for the boundary padding) and the edges of the overlay.
  // Note that these values are with respect to the visual viewport (aka 0,0 is the top left of the viewport)
  let boundaryStartEdge = boundaryDimensions.scroll[AXIS[axis]] + padding;
  let boundaryEndEdge = boundarySize + boundaryDimensions.scroll[AXIS[axis]] - padding;
// THE CONTAINER SCROLL BELOW SHOULD BE 0 IN THE CASE HIGHLIGHTED ABOVE SINCE OFFSET IS ALREADY IN THE SAME COORDINATE SYSTEM AS THE BOUNDARY VALUES.
  let startEdgeOffset = offset - containerScroll + containerOffsetWithBoundary[axis] - boundaryDimensions[AXIS[axis]];
  let endEdgeOffset = offset - containerScroll + size + containerOffsetWithBoundary[axis] - boundaryDimensions[AXIS[axis]];
```

Note that `startEdgeOffset = offset - containerScroll + containerOffsetWithBoundary[axis] - boundaryDimensions[AXIS[axis]];` should work as expected if the scroll of the `containerElement` is separate from the scroll of the `boundaryElement`